contracts/ETO/ETOTermsConstraints.sol

```diff
@@ -0,0 +1,132 @@
+    uint256 public constant MAX_PUBLIC_DURATION = 60 days;
+
+    // minimum length of whole offer
+    uint256 public constant MIN_OFFER_DURATION = 1 days;
+    // quarter should be enough for everyone
+    uint256 public constant MAX_OFFER_DURATION = 90 days;
+
+    uint256 public constant MIN_SIGNING_DURATION = 14 days;
+    uint256 public constant MAX_SIGNING_DURATION = 60 days;
+
+    uint256 public constant MIN_CLAIM_DURATION = 7 days;
+    uint256 public constant MAX_CLAIM_DURATION = 30 days;
+
+    // defines wether transfers are allowed after the eto ends
+    bool public CAN_SET_TRANSFERABILITY;
+
+    // defines wether a nominee is needed in the investment structure
+    bool public HAS_NOMINEE;
+
+    // minimum ticket size for this investment type
+    uint256 public MIN_TICKET_SIZE_EUR_ULPS;
+    // maximum ticket size for this investment type
+    uint256 public MAX_TICKET_SIZE_EUR_ULPS;
+    // minimum total investment amount this investment type
+    uint256 public MIN_INVESTMENT_AMOUNT_EUR_ULPS;
+    // maximum total investment amount this investment type
+    uint256 public MAX_INVESTMENT_AMOUNT_EUR_ULPS;
+
+    // public name
+    string public NAME;
+
+    // spec of the required offering document
+    OfferingDocumentType public OFFERING_DOCUMENT_TYPE;
+    OfferingDocumentSubType public OFFERING_DOCUMENT_SUB_TYPE;
+
+    // jurisdiction in which the ETO will be conducted
+    Jurisdiction public JURISDICTION;
+
+    // legal type of asset that will be used
+    AssetType public ASSET_TYPE;
+
+
+    ////////////////////////
+    // Constructor
+    ////////////////////////
+
+    constructor(
+        bool canSetTransferability,
+        bool hasNominee,
+        uint256 minTicketSizeEurUlps,
+        uint256 maxTicketSizeEurUlps,
+        uint256 minInvestmentAmountEurUlps,
+        uint256 maxInvestmentAmountEurUlps,
+        string name,
+        OfferingDocumentType offeringDocumentType,
+        OfferingDocumentSubType offeringDocumentSubType,
+        Jurisdiction jurisdiction,
+        AssetType assetType
+    )
+        public
+    {   
+        require(minTicketSizeEurUlps<=maxTicketSizeEurUlps);
+        require(minInvestmentAmountEurUlps<=maxInvestmentAmountEurUlps);
+        require(minTicketSizeEurUlps<=maxInvestmentAmountEurUlps);
+        require(maxTicketSizeEurUlps>0);
+        require(maxInvestmentAmountEurUlps>0);
+
+
+        CAN_SET_TRANSFERABILITY = canSetTransferability;
+        HAS_NOMINEE = hasNominee;
+        MIN_TICKET_SIZE_EUR_ULPS = minTicketSizeEurUlps;
+        MAX_TICKET_SIZE_EUR_ULPS = maxTicketSizeEurUlps;
+        MIN_INVESTMENT_AMOUNT_EUR_ULPS = minInvestmentAmountEurUlps;
+        MAX_INVESTMENT_AMOUNT_EUR_ULPS = maxInvestmentAmountEurUlps;
+        NAME = name;
+        OFFERING_DOCUMENT_TYPE = offeringDocumentType;
+        OFFERING_DOCUMENT_SUB_TYPE = offeringDocumentSubType;
+        JURISDICTION = jurisdiction;
+        ASSET_TYPE = assetType;
+    }
```

Comment:
would be cool to add simple validations:
if VMA asset type, CAN_SET_TRANSFERABILITY must be false (it's important from the regulatory perspective)

also how we encode "no limit" in case of MAX_TICKET_SIZE_EUR_ULPS and MAX_INVESTMENT_AMOUNT_EUR_ULPS, are those 0?

Reply:
I would say that no limit is int max

Reply:
previously they were set to maximum possible value of an ETO, I think we'll follow this rule when generating ETO TERMS